website/tools/minimize-language.js

```diff
@@ -5,9 +5,42 @@ import {
   existsSync,
 } from 'fs';
 
-for (const folder of readdirSync('./dist/locales', 'utf8')) {
-    if (existsSync('./dist/locales/' + folder + '/translation.json')) {
-      const content = readFileSync('./dist/locales/' + folder + '/translation.json', 'utf8',);
-      writeFileSync('./dist/locales/' + folder + '/translation.json', JSON.stringify(JSON.parse(content,),),);
+import * as yaml from 'yaml';
+import { dirname } from 'path';
+
+const generateJSONFromYAML = (yamlPath, outputPath) => {
+  if (existsSync(yamlPath)) {
+    try {
+      const content = readFileSync(yamlPath, 'utf8');
+      const jsonData = yaml.parse(content);
+      writeFileSync(outputPath, JSON.stringify(jsonData));
+    } catch {
+      console.log(`Error processing: ${yamlPath}`, error);
+    }
   }
-}
+};
+
+const originDir = 'language';
+const targetDir = 'public/locales';
+const yamlFiles = readdirSync('language');
+
+yamlFiles.forEach((yamlFile) => {
+  const lang = yamlFile.replace('.yml', '');
```

Comment:
this needs a filter for yaml files, otherwise you are also operating on `.` and `..`

Reply:
Oh right! I believe regex should've been used for that:)

Reply:
to make it easier you could use `file.endsWith('.yml')` instead of the regexp